### PR TITLE
Emit status code = 500 on error

### PR DESCRIPTION
**Pull request by mvorisek**
Industry standard and it seems that it can also catch errors in demos/tests.

### CODE REVIEW

**Comment by mvorisek** [comment]
Based on PR #1065. Rebase once that PR is merged.

Please help with fixing the test.

File: src/App.php [hunk 1]
```diff
@@ -302,32 +307,87 @@ public function outputDebug($str)
...
      *
      * @param string|array $output Array type is supported only for JSON response
-     * @param string       $contentType
+     * @param string[]     $headers Content-type header must be always set or consider using App::terminateHTML() or App::terminateJSON() methods.
      *
      * @throws \atk4\core\Exception
      * @throws ExitApplicationException
      */
-    public function terminate($output = null, string $contentType = 'text/html'): void
+    public function terminate($output = '', array $headers = []): void
```
Comment (romaninsh) [inline src/App.php]: @ibelar - will this break anything?

File: src/App.php [hunk 2]
```diff
@@ -987,47 +1074,90 @@ function () {
...
+            ob_end_flush();
+            self::$_x[] = [$headersNew, $trace, headers_list(), http_response_code()];
         }
 
-        echo $content;
+        if (count($headersNew) > 0 && headers_sent()) {
+            echo "\n" . '!! ATK4 UI ERROR: Headers already sent, more headers can not be set at this stage. !!' . "\n";
+
+            // debug
+            echo '--> Mvorisek:' . "\n";
```
Comment (romaninsh) [inline src/App.php]: should this be cleaned up?

**Review by romaninsh — APPROVED** [review]
I approve, but @ibelar should also review.

File: src/App.php (hunk 1 above)
Comment (mvorisek) [reply src/App.php]: It should not and null should never be allowed here. To not output anaything use nothing or empty string.

File: src/App.php (hunk 2 above)
Comment (mvorisek) [reply src/App.php]: @romaninsh Yes, it should, see my comment:

> Based on PR #1065. Rebase once that PR is merged.

So please review the PR #1065 first :)

**Comment by mvorisek** [comment]
Help wanted for the failing tests.

**Pull request merged**